keps/sig-auth/3257-trust-anchor-sets/README.md

```diff
@@ -0,0 +1,1158 @@
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+This KEP assumes that TLS and mutual TLS are technologies that we want to
+elevate to first-class status within the Kubernetes ecosystem.  This is a
+significant commitment, but not out of line with the existing decision to use
+OpenID Connect as the foundation for pod identity tokens.
+
+For the problems that TLS and mTLS address (non-exfiltratable identity
+certificates), there are no serious alternatives.
+
+## Alternatives
+
+### Do nothing in-tree; solve trust distribution with CRD/CSI driver
+
+This KEP is informed by the experience of implementing and operating
+`gke-spiffe`, an Kubernetes-integrated SPIFFE certificate issuance mechanism
+that backs several service mesh features.  It's built using the custom resource
+/ CSI driver approach, which works well.  However, there is appetite in SIG Auth
+for:
+
+1. Fixing parts of the Kubernetes API that embed static trusted root
+   certificates, like webhooks and aggregated API servers.  When these backends
+   use certificates issued by a private CA, rotating the root certificate of the
+   CA requires tracking down all objects that have these embedded roots and
+   updating them.  Adding a layer of indirection will allow rotation to be
+   performed with a single, central update.
+2. Making progress towards a future where pods are issued identity certificates,
+   just like they are issued identity tokens today.
+
+Both of these efforts will require a core object capable of describing trust
+anchors.
+
```

Comment:
There's also a hybrid option: CSI for the node-level part, but a core TrustAnchorSets API.

Reply:
I've updated this section of the doc to note that node-related stuff could be implemented using a CSI driver.